src/main/kotlin/com/ampnet/reportservice/service/data/TxSummary.kt

```diff
@@ -25,24 +30,36 @@ class TxSummary(
     val sharesBought = sumTransactionAmountsByType(TransactionsResponse.Transaction.Type.UNRECOGNIZED).toEurAmount()
     val sharesSold = sumTransactionAmountsByType(TransactionsResponse.Transaction.Type.UNRECOGNIZED).toEurAmount()
 
-    private fun getPeriod(transactions: List<Transaction>): String? {
+    private fun getPeriod(transactions: List<Transaction>, periodRequest: PeriodServiceRequest): String? {
+        val fromDate = periodRequest.from
+        val toDate = periodRequest.to
         return when (transactions.size) {
-            0 -> null
-            1 -> formatToYearMonthDay(transactions.first().date)
+            0 -> formatToYearMonthDay(fromDate ?: LocalDate.now())
```

Comment:
If it is easier to use LocalDate in `Transaction` class use it. You don't need to use `ZonedDateTime` because blockchain-service is not sending `ZonedDateTime` but `String` with value of epoch miliseconds.